src/engraving/libmscore/stretchedbend.cpp

```diff
@@ -261,26 +248,79 @@ void StretchedBend::layoutDraw(const bool layoutMode, mu::draw::Painter* painter
         switch (bendSegment.type) {
         case BendSegmentType::LINE_UP:
         {
-            if (layoutMode) {
-                bRect.unite(RectF(src.x(), src.y(), dest.x() - src.x(), dest.y() - src.y()));
-                bRect.unite(m_arrowUp.translated(dest).boundingRect());
+            painter->drawLine(LineF(src, dest));
+            painter->setBrush(curColor());
+            painter->drawPolygon(m_arrowUp.translated(dest));
+            /// TODO: remove substraction after fixing bRect
+            drawText(painter, dest - PointF(0, sp * 0.5), text);
+            break;
+        }
 
-                mu::draw::FontMetrics fm(font(sp));
-                bRect.unite(textBoundingRect(fm, dest, text));
-                /// TODO: remove after fixing bRect
-                bRect.setHeight(bRect.height() + sp);
-                bRect.setY(bRect.y() - sp);
-            } else {
-                painter->drawLine(LineF(src, dest));
-                painter->setBrush(curColor());
-                painter->drawPolygon(m_arrowUp.translated(dest));
-                /// TODO: remove substraction after fixing bRect
+        case BendSegmentType::CURVE_UP:
+        case BendSegmentType::CURVE_DOWN:
+        {
+            bool bendUp = (bendSegment.type == BendSegmentType::CURVE_UP);
+            double endY = dest.y() + m_bendArrowWidth * (bendUp ? 1 : -1);
+
+            PainterPath path = bendCurveFromPoints(src, PointF(dest.x(), endY));
+            const auto& arrowPath = (bendUp ? m_arrowUp : m_arrowDown);
+
+            painter->setBrush(BrushStyle::NoBrush);
+            painter->drawPath(path);
+            painter->setBrush(curColor());
+            painter->drawPolygon(arrowPath.translated(dest));
+
+            if (bendUp && !isTextDrawn) {
+                /// TODO: remove subtraction after fixing bRect
                 drawText(painter, dest - PointF(0, sp * 0.5), text);
+                isTextDrawn = true;
             }
 
             break;
         }
 
+        case BendSegmentType::LINE_STROKED:
+        {
+            PainterPath path;
+            path.moveTo(src + PointF(m_bendArrowWidth, 0));
+            path.lineTo(dest);
+            Pen p(painter->pen());
+            p.setStyle(PenStyle::DashLine);
+            painter->strokePath(path, p);
+            break;
+        }
+
+        default:
+            break;
+        }
+    }
+}
+
+void StretchedBend::layoutSegments()
```

Comment:
Why is the method called `layoutSegments`?
As far as I can see, it does nothing with segments.